### PR TITLE
Improve search API

### PR DESCRIPTION
**Pull request by lauraluiz**
Some improvements like better naming, better tests to be included in the documentation and correct date time implementation.

@schleichardt please review.

### CODE REVIEW

File: sphere-common/src/main/java/io/sphere/sdk/search/TypeSerializer.java [hunk 1]
```diff
@@ -75,8 +73,8 @@ private TypeSerializer(Function<T, String> serializer) {
      * Serializer to convert the given local datetime into the accepted ISO format with UTC zone.
      * @return the serializer for datetime data.
      */
-    public static TypeSerializer<LocalDateTime> ofDateTime() {
-        return new TypeSerializer<>(v -> withQuotes(v.atZone(UTC).format(ISO_DATE_TIME)));
+    public static TypeSerializer<ZonedDateTime> ofDateTime() {
+        return new TypeSerializer<>(v -> withQuotes(v.withZoneSameInstant(ZoneOffset.UTC).format(ISO_DATE_TIME)));
```
Comment (lauraluiz) [inline sphere-common/src/main/java/io/sphere/sdk/search/TypeSerializer.java]: #375 This works nicely as...

File: sphere-common/src/test/java/io/sphere/sdk/search/TypeSerializerTest.java [hunk 2]
```diff
@@ -54,8 +54,8 @@ public void serializesTime() throws Exception {
 
     @Test
     public void serializesDateTime() throws Exception {
-        Function<LocalDateTime, String> serializer = ofDateTime().serializer();
-        assertThat(serializer.apply(dateTime("2001-09-11T22:05:09.203"))).isEqualTo("\"2001-09-11T22:05:09.203Z\"");
+        Function<ZonedDateTime, String> serializer = ofDateTime().serializer();
+        assertThat(serializer.apply(dateTime("2001-09-11T22:05:09.203+02:00"))).isEqualTo("\"2001-09-11T20:05:09.203Z\"");
```
Comment (lauraluiz) [inline sphere-common/src/test/java/io/sphere/sdk/search/TypeSerializerTest.java]: #375 ... this.

**Comment by lauraluiz** [comment]
Solves #338

**Comment by schleichardt** [comment]
Good, thanks.

**Pull request merged**